Ttrack 105

zerocity opened the pull request:
- Implement Error,Request Logging with a human readable time format
- Fix Error logging with logger
- change console log to server loggging 
- remove unused handler 
- fix user test
- Add swagger documentation
- Add swagger dependencies to manifest

Commit:
Improve Log messages

- Enable logging for log, error, response tags 
- change Datetime format

Commit:
Fix Error Promise - error logging

console.error blocks  routing  the error message  to the logger

Commit:
Implement Hapi-swagger

- Add swagger documentation
- Add swagger dependencies to manifest
- change console log to server loggging 
- remove unused handler 
- fix user test

dready commented on src/api/handlers/period.js:
```diff
@@ -1,85 +1,120 @@
+    plugins: {
+        'hapi-swagger': {
+            responses: {
+                '200': {
```
shoudl be http status code 201 = created

dready commented on src/api/handlers/user.js:
```diff
@@ -1,34 +1,65 @@
+                    description: 'The user with the corresponding id',
+                    schema: User
+                },
+                '400':{
```
should be 404 - not found instead of just 400

dready commented on package.json:
```diff
@@ -21,6 +21,7 @@
     "good-squeeze": "^5.0.2",
     "hapi": "^16.4.3",
     "hapi-405-routes": "^0.6.1",
+    "inert": "^4.2.1",
```
do we need inert and vision for the swagger output?

dready reviewed (commented):
great swagger introduction, just a few minor details

Commit:
Update Swagger documentation Status codes

Commit:
Update Status codes

Commit:
Add 404 Tests
- add 404 status code messages
- remove promise error handling in resources
- imporve wallaby config

dready commented on src/api/__test__/api-period.test.js:
```diff
@@ -434,15 +479,6 @@ describe('ttrack API',() => {
                 expect(result.rows.length).toBe(0);
             });
-
-            it("should not throw any errors if period is unknown", async () => {
```
why did you remove that?
Delete on a Period that is not there should be idempotent, therefore always returning 204

dready commented on src/api/handlers/period.js:
```diff
@@ -114,7 +117,10 @@ module.exports.delete = {
         return PeriodResources
             .delete(per_id, userId)
             .then(
-                () => reply().code(204)
```
nope, the call should be idempotent. Think it that way. There is a period. You send delete on it. It will return 204. You call delete again, the outcome is the same, there is no period there afterwards, therefore it should be 204

dready reviewed (commented):
great change. only the delete resource is not right. it should always return 204, even if the period is not available. Only if the user is unknown a 404 would be correct.

Commit:
Fix Status Codes

- remove 404 status codes on wrong period ids 
- recover old period test
